spec/requests/dashboard_spec.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Dashboard', type: :request do
+  describe '#reports' do
+    context 'as an admin' do
+      let(:user) { create :user, role: 'admin' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_reports_path
+  
+        expect(response).to be_successful
+      end
+    end
+
+    context 'as a coordinator' do
+      let(:user) { create :user, role: 'coordinator' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_reports_path
+  
+        expect(response).to be_successful
+      end
+    end
+
+    context 'as a volunteer' do
+      let(:user) { create :user, role: 'volunteer' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_reports_path
+  
+        expect(response).to redirect_to :root
+      end
+    end
+
+    context 'as a social worker' do
+      let(:user) { create :user, role: 'social_worker' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_reports_path
+  
+        expect(response).to be_successful
+      end
+    end
+  end
+
+  describe '#users' do
+    context 'as an admin' do
+      let(:user) { create :user, role: 'admin' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_users_path
+  
+        expect(response).to be_successful
+      end
+    end
+
+    context 'as a coordinator' do
+      let(:user) { create :user, role: 'coordinator' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_users_path
+  
+        expect(response).to be_successful
+      end
+    end
+
+    context 'as a volunteer' do
+      let(:user) { create :user, role: 'volunteer' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_users_path
+  
+        expect(response).to redirect_to :root
+      end
+    end
+
+    context 'as a social worker' do
+      let(:user) { create :user, role: 'social_worker' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_users_path
+  
+        expect(response).to redirect_to :root
+      end
+    end
+  end
```

Comment:
@benjaminwood I don't like this repeating pattern for checking authorization, but I dislike _not_ explicitly testing it more. In this case where we have a policy and spec that covers these scenarios already, is it worth to check at the controller level as well? My gut says yes.